More update on the guidance

cccclai opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72818

Differential Revision: [D34226823](https://our.internmc.facebook.com/intern/diff/D34226823/)

Commit:
More update on the guidance



Differential Revision: [D34226823](https://our.internmc.facebook.com/intern/diff/D34226823/)

[ghstack-poisoned]

raziel commented on torch/csrc/jit/operator_upgraders/README.md:
```diff
@@ -1,8 +1,50 @@
-PyTorch’s operators sometimes require changes to maintain the high quality user experience (UX) that PyTorch is known for. These changes can be backward compatibility (BC) breaking, where older programs will no longer run as expected on the latest version of PyTorch (an old writer / new reader problem) or forward compatibility (FC) breaking, where new programs will not run on older versions of Py ...
+PyTorch’s operators sometimes require changes to maintain the high quality user experience (UX) that PyTorch is known for. These changes can be backward compatibility (BC) breaking, where older programs will no longer run as expected on the latest version of PyTorch (an old program / new runtime problem) or forward compatibility (FC) breaking, where new programs will not run on older versions of  ...
+
+If the change to the operator is BC-breaking in either the schema or the semantics way, you will need to write an upgrader to make the change non-BC breaking. In general, you can know your operator is BC breaking, if it fails `test/forward_backward_compatibility/check_forward_backward_compatibility.py `.
```
```suggestion
If the change to the operator is BC-breaking in either the schema or the semantics, you are responsible for writing an upgrader to prevent the change from becoming BC breaking.

You can determine if your change in the operator is BC breaking, if it fails `test/forward_backward_compatibility/check_forward_backward_compatibility.py `.
```

raziel commented on torch/csrc/jit/operator_upgraders/README.md:
```diff
@@ -1,8 +1,50 @@
+
+### Some examples BC/FC breaking changes
+
+When making changes to the operators, the first thing to identify is if it's BC/FC breaking. Again, we only targetting for BC breaking changes on this guidance. Here are some examples to help understanding what a BC/FC changes may look like:
```
```suggestion
When making changes to the operators, the first thing to identify is if it's BC/FC breaking. Again, we only targetting for BC breaking changes on this guidance. Here are some examples to help understanding what a BC/FC changes may look like:
```

raziel commented on torch/csrc/jit/operator_upgraders/README.md:
```diff
@@ -1,8 +1,50 @@
+
+If the change to the operator is BC-breaking in either the schema or the semantics way, you will need to write an upgrader to make the change non-BC breaking. In general, you can know your operator is BC breaking, if it fails `test/forward_backward_compatibility/check_forward_backward_compatibility.py `.
+
+### Some examples BC/FC breaking changes
```
should we even include FC breaking here? maybe not since we don't have any solution for it and it may confuse people more than help.

cccclai replied:
I move the FC breaking example to the note at the end. My intention is to make it easy for users to categorize their change. Let me know if it makes sense

Commit:
Update on "More update on the guidance"


Differential Revision: [D34226823](https://our.internmc.facebook.com/intern/diff/D34226823/)

[ghstack-poisoned]

raziel reviewed (approved):
Thanks for the changes.

It'd be great to find somebody that needs to write an upgrader that goes over these instructions and update as needed.

@cccclai can you please follow up to identify a candidate? Thanks

Commit:
Update on "More update on the guidance"


Differential Revision: [D34226823](https://our.internmc.facebook.com/intern/diff/D34226823/)

[ghstack-poisoned]
